types.h

```diff
@@ -5,6 +5,9 @@
 
 /*Define some useful types*/
 
+#define MPI_UINT64 MPI_UNSIGNED_LONG
+#define MPI_INT64 MPI_LONG
```

Comment:
I recall in a few places MPI_LONG and MPI_LONG_LONG are still used instead MPI_INT64 / MPI_UINTT64.

Perhaps also define MPI_PTRDIFF and MPI_SIZET and replace them all?

Reply:
Should it be #define MPI_INT64 MPI_LONG_LONG ? I don't think MPI_LONG is guaranteed to be 64-bit

Reply:
I will leave it as a comment - we are probably more likely to run on a 128-bit architecture than a 32-bit one right now.